app/src/main/java/org/jellyfin/androidtv/ui/search/SearchRepository.kt

```diff
@@ -0,0 +1,45 @@
+package org.jellyfin.androidtv.ui.search
+
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.itemsApi
+import org.jellyfin.sdk.model.api.BaseItemDto
+import org.jellyfin.sdk.model.api.BaseItemKind
+import org.jellyfin.sdk.model.api.ItemFields
+import java.io.IOException
+
+interface SearchRepository {
+    suspend fun search(searchTerm: String, itemTypes: Collection<BaseItemKind>): List<BaseItemDto>
+}
+
+class SearchRepositoryImpl(
+        private val apiClient: ApiClient
+) : SearchRepository {
+    companion object {
+        private const val QUERY_LIMIT = 25
+    }
+
+    override suspend fun search(
+            searchTerm: String,
+            itemTypes: Collection<BaseItemKind>
+    ): List<BaseItemDto> {
+        return try {
+            val result = apiClient.itemsApi.getItemsByUserId(
+                    searchTerm = searchTerm,
+                    limit = QUERY_LIMIT,
+                    imageTypeLimit = 1,
+                    includeItemTypes = itemTypes,
+                    fields = listOf(
+                            ItemFields.PRIMARY_IMAGE_ASPECT_RATIO,
+                            ItemFields.CAN_DELETE,
+                            ItemFields.BASIC_SYNC_INFO,
+                            ItemFields.MEDIA_SOURCE_COUNT
+                    ),
+                    recursive = true,
+                    enableTotalRecordCount = false,
+            )
+            result.content.items ?: emptyList()
+        } catch (e: IOException) {
```

Comment:
Need to double check that IOException is what gets thrown on an http error

Otherwise we can catch `Exception`, if you'd rather just do that

Reply:
Nevermind, looks like `ApiClientException` is what we want